gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -131,9 +134,28 @@ def kerchunkify(self, file_path: str, scan_indices: int = 0):
                 self.zarr_jsons.append(scanned_zarr_json)
             elif type(scanned_zarr_json) == list:
                 self.zarr_jsons.extend(scanned_zarr_json)
+        # output individual JSONs for re-reading locally. This guards against crashes for long Extracts and speeds up dev. work.
+        if self.write_local_zarr_jsons:
+            if isinstance(scanned_zarr_json, list):  # presumes lists are not nested more than one level deep
+                memory_write_args = zip(scanned_zarr_json, repeat(local_file_path))
+                list(starmap(self.zarr_json_in_memory_to_file, memory_write_args))
+            else:
+                self.zarr_json_in_memory_to_file(scanned_zarr_json, local_file_path)
 
         return scanned_zarr_json
 
+    def zarr_json_in_memory_to_file(self, scanned_zarr_json: str, local_file_path: str):
+        """
+        Export a Kerchunked Zarr JSON to file.
+        Implementations of this method rely on 
+        """
+        if hasattr(self, "data_driven_zarr_json_name"):
```

Comment:
My feeling is we want to avoid hasattr as much as possible. In this case, I think having a base class `DatasetManager.data_driven_zarr_json_name` function that just returns its input and then overriding it in CFS would be a way to do the same thing without needing hasattr.